gravitee-am-service/src/main/java/io/gravitee/am/service/authentication/crypto/password/MessageDigestPasswordEncoder.java

```diff
@@ -66,9 +75,20 @@ public String encode(CharSequence rawPassword) {
     @Override
     public String encode(CharSequence rawPassword, byte[] salt) {
         try {
-            MessageDigest md = MessageDigest.getInstance(algorithm);
-            md.update(salt);
-            byte[] hashedPassword = md.digest(rawPassword.toString().getBytes(StandardCharsets.UTF_8));
+            byte[] hashedPassword;
+            if (PasswordSaltFormat.DIGEST.equals(passwordSaltFormat)) {
+                MessageDigest md = MessageDigest.getInstance(algorithm);
```

Comment:
As we update the messageDigest object (update(salt) method), I would prefer to create a new instance and keep the same behavior as before to avoid any breaking changes.